From 8df2234a3cd105b84e8f3d00c97ed6226c1fa713 Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@apache.org>
Date: Thu, 26 Oct 2017 17:41:53 -0700
Subject: [PATCH 2813/2863] HDFS-12710. HTTPFS HTTP max header size env
 variable is not respected in branch-2.
 Contributed by Istvan Fajth.

(cherry picked from commit 952aa3f168eb485e1fe7f045ab133399f6d626dc)

Change-Id: I243d7d50d3540ab47f9bed2bf6a3198c10553cae
(cherry picked from commit 7df670f04dd3d691efa3e4979692b01a35ec1ca2)
---
 .../hadoop-hdfs-httpfs/src/main/sbin/httpfs.sh     |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs-httpfs/src/main/sbin/httpfs.sh b/hadoop-hdfs-project/hadoop-hdfs-httpfs/src/main/sbin/httpfs.sh
index 75b398e..8accc59 100644
--- a/hadoop-hdfs-project/hadoop-hdfs-httpfs/src/main/sbin/httpfs.sh
+++ b/hadoop-hdfs-project/hadoop-hdfs-httpfs/src/main/sbin/httpfs.sh
@@ -67,6 +67,8 @@ if [[ "${1}" = "start" || "${1}" = "run" ]]; then
   catalina_set_property "httpfs.admin.port" "${HTTPFS_ADMIN_PORT}"
   catalina_set_property "httpfs.http.port" "${HTTPFS_HTTP_PORT}"
   catalina_set_property "httpfs.http.hostname" "${HTTPFS_HTTP_HOSTNAME}"
+  catalina_set_property "httpfs.max.http.header.size" \
+    "${HTTPFS_MAX_HTTP_HEADER_SIZE}"
   catalina_set_property "httpfs.ssl.enabled" "${HTTPFS_SSL_ENABLED}"
   catalina_set_property "httpfs.ssl.client.auth" "${HTTPFS_SSL_CLIENT_AUTH}"
   catalina_set_property "httpfs.ssl.enabled.protocols" \
-- 
1.7.9.5

